rfcs/0112-demote-darwin-support-tier.md

```diff
@@ -0,0 +1,91 @@
+---
+feature: demote_darwin_support_tier
+start-date: 2021-10-27
+author: piegames
+co-authors: many
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
```

Comment:
I'd like to nominate @risicle, who is very aware of the issues around Darwin in nixpkgs.

Reply:
I think I'd be very conflicted.

As much as part of me wants to cast it into the pits of hell, I've come to the realization that what with homebrew coming to linux it's becoming a squash-or-be-squashed situation. Broad strokes here, but either the linux crowd present the macos developer community with a more palatable alternative or linux users will find themselves expected to use homebrew. I've already encountered situations joining teams where all the instructions and tooling for getting a development environment up and running expect homebrew, and it's only something that's going to grow. So my interest would be in how we can grow the number of darwin developers instead of demoting darwin.

That said, I think this is going to become academic because Apple will likely make macos an unworkable platform within a release or two.

Reply:
I'd like to nominate @andir

Reply:
Since I'm apparently nominated as of https://github.com/NixOS/rfc-steering-committee/issues/77: I would prefer not to if we can find enough other persons. Quite a few people helped write this RFC by providing their expertise and experiences. It really showed me how little I actually know about the topic compared to them.

Reply:
That seems to be a typo. By the process in #36 the author(s) is never among the shepherds (but usually included in the shepherd meetings)

Reply:
Yes, apologies! That was a misread on my part

Reply:
I've seen this misconception a few times now, so I'd like to clarify: You do *not* need to be in favor of the proposed changes in order to shepherd this RFC. Ideally, the set of shepherds would be as representative for the different positions and opinions voiced during the discussion as possible.

If the shepherds decide that the RFC needs an overhaul or rewrite, or that we should try a different approach than the one proposed here, then this is totally fine.

Reply:
I am happy to accept the nomination and participate an the shepharding of this RFC. In whatever direction this will move this discussion and finally an actionable RFC. My position on this is two fold: I'm just as much in favor of this RFC (primarily for the discussions sake) but I see how Darwin has a special role. My primary object would be to find a better formal way to deal with platform incompatibilities and perhaps a sunsetting plan for x86_64-darwin as the end of hardware sales is near.

Reply:
Is `x86-linux` a Tier 1 platform?

Reply:
@dhess `i686-linux` is not strictly speaking a Tier-1 platform but large parts of it are effectively treated at almost Tier-1 standards because they are dependencies of Wine build for the Tier-1 `x86_64-linux` platform.

Reply:
It's gotten a bit quiet in here. But we are still looking for more shepherds in order to move forward.